packages/terra-date-input/src/DateInput.jsx

```diff
@@ -442,6 +468,36 @@ class DateInput extends React.Component {
     }
   }
 
+  /**
+   * Sets state to new value dependent on existing date
+   * @param {Object} event Event object generated from the event delegation.
+   * @param {Number} addDays Adds days to current date or today's date.
+   */
+  setHotKeyDate(event, addDays) {
+    const { year, month, day } = this.state;
+    // The Date constructor uses 0-based index for month, using our set month value will give the next month but by also using 0 for day
+    // we move the day back by one giving the set month's final day.
```

Comment:
This is too clever to be maintained I think. It's very tricky and seems likely to be broken in the future when another person has to fix, reason-about, etc. Even if it works... I'm not comfortable with this.

Reply:
Use Momentjs, and you'll have a one-liner that is dead easy to understand and far less risky to maintain. https://momentjs.com/docs/#/parsing/is-valid/

```
moment().isValid();
```

Reply:
I mean specifically the momentjs w/o the timezones: ~ 22kb

Reply:
@BenBoersma showed me offline that it's not as simple as using momentjs.isValid() because that doesn't really check if the user entered month 35.

Here is another tactic: Since a valid date should parse to itself, but an invalid date will parse to something else, try comparing the date generated by the inputs back against the original input values like so: https://codepen.io/elliott-hoffman-cerner/pen/jOGBYVb

Reply:
Okay reworked the logic here: https://github.com/cerner/terra-framework/pull/1483/commits/fe464dfb64ddf3f72184bfea172d374d81b17473